### DISCUSSION TITLE
Ways to improve performance?

### DISCUSSION THREAD

**Discussion by GavanWilhite** [vercel/satori#431] [Q&A]
I can't seem to get faster performance than about 1.5 seconds locally and 2.5 seconds deployed to vercel. 

It's just some basic text from a custom font, and a few emoji.

All user code including font loading takes 15ms cold and less than 1 hot, so I'm pretty sure it must be rendering or something else. 

Thoughts?

**Comment by shuding** [comment]
Can it be related to the emojis? FWIW .ttf font with only necessary charsets (e.g. Latin) and inlined emoji images (if they are static) will make it a bit faster.

**Comment by GavanWilhite** [comment]
This is interesting. I'll have to look into this

**Comment by shuding** [comment]
You can analyze what % of time was spent on fetching fonts and images. And then, in general smaller and simpler images (fewer pixels) will be much faster.

**Comment by GavanWilhite** [comment]
Can I do this if I'm using ImageResponse with Vercel? I think it happens
behind the scenes.
